se/se_epub_generate_toc.py

```diff
@@ -328,27 +354,38 @@ def output_toc(item_list: list, landmark_list, toc_path: str, work_type: str, wo
 	if len(item_list) < 2:
 		raise se.InvalidInputException("Too few ToC items found.")
 
-	existing_toc: BeautifulSoup = get_existing_toc(toc_path)
-	if existing_toc is None:
+	try:
+		with open(toc_path) as file:
+			toc_dom = se.easy_xml.EasyXhtmlTree(file.read())
+	except Exception as ex:
 		raise se.InvalidInputException("Existing ToC not found.")
 
 	# There should be exactly two nav sections.
-	navs = existing_toc.find_all("nav")
+	navs = toc_dom.xpath("//nav")
 
 	if len(navs) < 2:
 		raise se.InvalidInputException("Existing ToC has too few nav sections.")
 
-	item_ol = navs[0].find("ol")
-	item_ol.clear()
-	landmark_ol = navs[1].find("ol")
-	landmark_ol.clear()
-	new_items = BeautifulSoup(process_items(item_list), "html.parser")
-	item_ol.append(new_items)
-	new_landmarks = BeautifulSoup(process_landmarks(landmark_list, work_type, work_title), "html.parser")
-	landmark_ol.append(new_landmarks)
-	return format_xhtml(str(existing_toc))
-
-def get_parent_id(hchild: Tag) -> str:
+	# now remove and then re-add the ol sections to clear them
+	for nav in navs:
+		ols = nav.xpath("ol")  # just want the immediate ol children
```

Comment:
If you want the immediate `ol` children of `nav`, then the xpath should be: `nav.xpath("./ol")`